api4/user.go

```diff
@@ -1191,6 +1193,64 @@ func generateMfaSecret(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(secret.ToJson()))
 }
 
+func generateRecovery(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if c.App.Session.IsOAuth {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		c.Err.DetailedError += ", attempted access by oauth app"
+		return
+	}
+
+	if !c.App.SessionHasPermissionToUser(c.App.Session, c.Params.UserId) {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
```

Comment:
@jespino Was the decision to allow system admins to create recovery tokens for other users intentional?

Reply:
I'm not sure, but I think it follows the same pattern found in other places, like be able to reset the password. I have not an strong opinion about this.